qubes/tests/vm/qubesvm.py

```diff
@@ -1439,7 +1439,7 @@ def test_610_libvirt_xml_network(self):
                 <script path="vif-route-qubes" />
             </interface>
             <!-- server_ip is the address of stubdomain. It hosts it's own DNS server. -->
-            <emulator type="stubdom-linux" />
+            <emulator type="stubdom-linux" cmdline="-qubes-net:client_ip=10.137.0.1,dns_0=10.139.1.1,dns_1=10.139.1.2,gw=10.137.0.2,netmask=255.255.255.255" />
```

Comment:
It may not matter since this is a test, but netmask changed from 255.255.255.0 in [linux-stubdom](https://github.com/pwmarcz/qubes-vmm-xen-stubdom-linux/commit/53648843f698cf45f1e5d556d46976dfd025a31b#diff-9b7847ba61b5a8a5e3ecc2a7952b1c3fL36) to 255.255.255.255 here.

Reply:
Ah, indeed, thanks! Generally this is what we set in Linux VMs, together with explicit route to a gateway, but AFAIR Windows is not happy about it.
We have proxy_arp enabled, so exact netmask doesn't matter anyway (connected network-providing vm will respond with its own MAC for ARP requests for routed addresses too),

BTW Where (if at all) do you run DHCP server in OpenXT?

Reply:
DHCP runs inside the Network Device VM (ndvm).  ndvm has 5 bridges: brbridged, brinternal0, brwireless, brshared & brany.  brany, brshared & brwireless provide DHCP & NAT for guests.  brshared is for wired, brwireless is wireless, and brany will use either.  brbridged just bridges guests onto the wired ethernet, and brinternal is a disconnected bridge.

Why do you run DHCP out of stubdom instead of sys-firewall or sys-net?  Do you also run DHCP for the Xen vifs or push their IPs in through QubesRPC?

Reply:
> Why do you run DHCP out of stubdom instead of sys-firewall or sys-net? 

To not extend attack surface of them, since they see traffic of other VMs and also apply policies (firewall).

> Do you also run DHCP for the Xen vifs or push their IPs in through QubesRPC?

Not qubes-rpc specifically, but a qubes-specific startup script, which gets info from qubesdb (xenstore-like thing).

Reply:
Thanks!